### DISCUSSION TITLE
Trouble running TCPairs basic use case

### DISCUSSION THREAD

**Discussion by mkavulich** [dtcenter/METplus#2604] [Use Cases] [answered]
I am having trouble running the [TCPairs basic use case for tropical cyclones](https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/TCPairs/TCPairs_tropical.html#tcpairs-basic-use-case-for-tropical-cyclones). The conf file provided for running the example case does not correspond to the sample data provided (https://dtcenter.ucar.edu/dfiles/code/METplus/METplus_Data/v5.1/sample_data-tc_and_extra_tc.tgz): the dates and storms in the `.conf` file are from 2018, and the sample data is from 2021.

When I change the dates and storm variables manually I do get a bit further, but then I hit another barrier: I see references in the conf file to a "gen_dland" utility that should be producing a file `dland_global_tenth_degree.nc`, but I do not see any references in the users guide to how I should use this utility (or where to even find it). This file is not included in the sample data either, and I'm having trouble finding any references anywhere on the internet to this mysterious tool and file.

I am running on Jet and referencing the METplus build that was installed by EPIC here: `/mnt/lfs4/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.5.1/envs/unified-env-rocky8/install/intel/2021.5.0/metplus-5.1.0-qogotil`

Let me know if there's any other information I can provide.

**Comment by j-opatz** [comment, marked as answer]
Hello,

And thank you for your question. I think the solution to this issue might be a better explanation of the use case you're trying to get to run. The "basic use cases", demonstrating METplus wrapper usage (including the TCPairs_tropical configuration file you linked to) are located in the met_tool_wrapper directory of the repository. [The link here](https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/index.html#) shows all of these configuration files gathered together, under the common category name "MET tools". As such, these examples rely on data located in [met_test](https://dtcenter.ucar.edu/dfiles/code/METplus/METplus_Data/v5.1/met_test/), a separate directory from the data necessary to run the other use cases, which are gathered under the common category name "Model Applications". So the link you provided to the sample data for tc and extra tc does not contain the necessary dataset for the TCPairs_tropical met_tool example to run properly; it contains data for the model applications use cases and you need the dataset for MET tools. You correctly saw that the configuration file needs data from 2018 while the data in the .tgz file contains data from 2021.

A quick way to tell which data directory the correct datasets are in is to check the _INPUT_DIR variables of the configuration file. For the TCPairs_tropical configuration file, it's 

```
TC_PAIRS_ADECK_INPUT_DIR = {INPUT_BASE}/met_test/new/hwrf/adeck
TC_PAIRS_BDECK_INPUT_DIR = {INPUT_BASE}/met_test/new/hwrf/bdeck
```
Note how both variable settings start with the user's {INPUT_BASE}, but are immediately followed by "met_test": this says the correct dataset will be in the met_test data folder. On JET, I see that all test data (for both "MET tools" and "Model Applications") should be located in /lfs1/HFIP/dtc-hurr/METplus/METplus-5.1_sample_data. Setting this path for your {INPUT_BASE} with no changes to the configuration file should correctly pull from the met_test directory instead of the model_applications. If you need help with setting this, please let me know.

For a little more background context, the biggest difference between these two categories of use cases, "MET tool" and "model applications", is that the MET tool category serves as a example of how to run the wrapped tool only; it is not meant to produce meaningful data, nor does it represent any scientific meaningfulness aside from how to use the tool. The model applications category contains examples of how one or more wrapped tools were used to answer scientific objectives, using real data and producing statistics that were useful to the parties involved with creating the use case. As such, it made sense to separate these two data categories into separate directories.

**Comment by j-opatz** [comment]
Hi @mkavulich,

I wanted to reach out again to see if you had any further questions related to this Discussions topic. If not, I will select an answer and lock the topic from any further comments.

**Reply by mkavulich** [reply]
@j-opatz Thanks for the replies, sorry I haven't gotten around to trying your solution yet. I was planning to try again today but unfortunately the Jet file system I need (lfs4) is down and there is no ETA for restoration.

As one follow-up question, does this mean that if I do need to download sample data for these "Model Application" cases, I'll need to `wget` or in some other way retrieve the data from the https://dtcenter.ucar.edu/dfiles/code/METplus/METplus_Data/v5.1/model_applications/ URL and subdirectories? It doesn't seem as if there's a corresponding tarball for downloading data for these model application cases.

**Reply by j-opatz** [reply]
There is an easier way, luckily! If you go to the parent directory of the link you provided ([this one here](https://dtcenter.ucar.edu/dfiles/code/METplus/METplus_Data/v5.1/)) you'll see that all of the "sample_data_xyz" tgz files are available. These will contain data for all of the model application use cases of that specific type. 

For example, "sample_data-precipitation.tgz" contains all of the data to run [the 10 precipitation-type use cases](https://metplus.readthedocs.io/en/latest/generated/model_applications/index.html#precipitation).

In a similar way, "sample_data-met_tool_wrapper-5.1.tgz" contains all of the files necessary to run the "met tool wrapper" examples. This tgz would contain the necessary data for the TC example you were trying to run.

**Comment by j-opatz** [comment]
Hi @mkavulich,

I wanted to reach out and see if the above suggestions have answered your questions. If it has, please mark an answer and I'll go ahead and lock the question from further discussions.

If you still have questions related to this topic or the suggestions didn't quite solve the issue, please reach out and let me know!

**Comment by mkavulich** [comment]
Thanks for your help @j-opatz, I was able to run the case successfully once I was able to use the correct data.

While I have you, I did notice a minor error in the basic TCPairs use case: the provided conf file sets `TC_PAIRS_DLAND_FILE`, but this was not the correct setting; I had to instead set `METPLUS_DLAND_FILE` to get the run to complete. Should I open a separate issue for this correction?

**Reply by j-opatz** [reply]
That's a good question I'd like to look into with a separate Discussions topic. I've gone ahead and created a new one here: https://github.com/dtcenter/METplus/discussions/2625
